spinn_machine/sdram.py

```diff
@@ -17,6 +18,7 @@ def __init__(self, size=DEFAULT_SDRAM_BYTES):
         if size < 0:
             raise SpinnMachineInvalidParameterException(
                 "size", size, "negative sizes are meaningless")
+        SDRAM.max_sdram_found = max(SDRAM.max_sdram_found, size)
```

Comment:
this should be put in machine and have a safety check that its consistent over the machine. otherwise the one place this value is used is actually not being a test but a "maybe test". If its not consistent ove rthe machine, the better solution is to give the chip its being placed on somewhere in the dsg, or remove the safety check entirely.